### PR TITLE
Docusaurus switch

### LINKED ISSUE #649: Convert to Docusaurus and “compile” all documentation
Currently we have:
* [this](https://github.com/packit/packit.dev) repository containing weekly posts, docs, blog posts
* [research](https://github.com/packit/research) repository for spikes and arch/design decisions
* [deployment](https://github.com/packit/deployment/tree/main/docs/) docs
* [contributing](https://github.com/packit/contributing) guidelines
  * additionally repo-specific contribution guidelines

This setup is fragmented and not very intuitive, since the extensive documentation is spread across multiple repositories (or even directories within one repository).

TODO:
- [ ] Switch to Docusaurus -- this allows us to separate the docs into multiple sections; also brings more functionality compared to Hugo (regarding the formatting, etc.)
- [ ] »Slowly« migrate fragments into one documentation that can be used both internally by the team and also by the users
  - [ ] There are issues with links and also handling of the images, preview below is MVP

Preview: https://me.mfocko.xyz/packit
([source](https://github.com/packit/docusaurus-mockup-for-retro) for the preview; internal-only)

### PR DESCRIPTION
**Pull request by mfocko**
Fixes #649 

* Live preview: https://me.mfocko.xyz/packit

TODO:

- [x] Front page!
- [x] Documentation
  - [x] Check any Hugo-specific features
  - [x] Convert Hugo details to HTML details
  - [x] Should be done, ready for “rip and tear until it's done”
- [x] Source-git
  - [x] Check…
  - [x] Hugo links
- [x] Deployment
  - [x] Merge the changes back into the deployment repo
  - [ ] packit/deployment#507
- [x] Research
  - [x] Consider including it; probably follow-up PR, since it requires more changes
- [x] Blog posts
  - [x] Check…
  - [x] Polish off the “previews”
    - [x] Definitely remove the headings
- [x] Weekly updates
  - [x] Check…
  - [ ] Consider switching to post per week
  - [ ] Make sure the RSS for weekly updates is updated in the linked Fedora digest or something! ask @xsuchy 
- [ ] Shortcodes redo
  - [ ] Architecture
  - [ ] Usage charts
  - [ ] Workflow
- [x] Footer
  - [ ] Decide what to include: Docusaurus themselves link back the docs, community/contacts, github/mastodon and legal
- [x] Navbar
  - [x] Probably link back to GitHub, dashboard — basically former external links on the left panel in Hugo
- [ ] Check 1:1 links against Hugo
- [x] GitHub Action for building (and deploying on merge)
- [x] Remove stuff from Docusaurus template
- [x] Consider changing the CSS, e.g. Packit colors instead of the Docusaurus green one
- [x] Finally! Move out from the nested directory that simplifies rebasing :cold_sweat: 
- [x] Resolve broken links reported during build
- [ ] Set up [search](https://docusaurus.io/docs/search#using-algolia-docsearch)

### CODE REVIEW

**Review by TomasTomecek — APPROVED** [review]
it's true that the panel "breaks" after zooming, but it's not such a big deal to me, I often zoom in/out sites.

We can fix this specifically after merging & deploying with CSS.

File: yarn.lock
Comment (majamassarini) [inline yarn.lock:1]: Probably this should not be tracked

File: README.md [hunk 1]
```diff
@@ -1,77 +1,189 @@
...
+[packit.dev] website.
 
-## Clone
+### Installation
 
-This repository uses git submodules, so you have to `git clone --recurse-submodules` it.
-If you forget and realize later, just run:
+```
+$ yarn
+```
```
Comment (majamassarini) [inline README.md:10]: I am trying to work with it locally.
I didn't realize immediately that `yarn` is not a `dnf` package. I installed it through `npm` but I was wondering if this is our preferred way? Maybe we can write it down.

File: README.md [hunk 2]
```diff
@@ -1,77 +1,189 @@
...
 
-    $ git submodule init themes/book
-    $ git submodule update
+### Local Development
 
-## Hugo
+```
+$ yarn start
+```
```
Comment (majamassarini) [inline README.md:16]: This is not working for me, it tells me:

```[ERROR] Error: The docs folder does not exist for version "current". A docs folder is expected to be found at research```

Should we introduce a makefile target with other steps before this one? Am I missing something?

**Review by majamassarini — COMMENTED** [review]
Thanks Matej, I think this is really a big improvement!

I like the new look&feel and I have 3 small tips (more some questions related with the "code").

1. I was looking for a way to search through the new site but I was not able to find a search feature. I think we should have it.
2. I find the new front page much nicer but I would like to insert some more space between the blue banner and the content on the white space (I tried to hack it, but I am not able to run the server locally and test my suggestions...)
3. What about moving the `contact` paragraph in the front page below, inside the dark gray banner under the `community` column. In this way we should have the contact always visible in every page?

File: yarn.lock
Comment (mfocko) [reply yarn.lock:1]: It's used here:
https://github.com/mfocko/packit.dev/blob/docusaurus/.github/workflows/deploy.yml#L22C2-L23

And also in the test build on the PRs. It makes sure it gets built with the same versions everywhere, i.e. if some of the dependencies get updated and break the build, it shouldn't affect the CI.

File: README.md (hunk 1 above)
Comment (mfocko) [reply README.md:10]: Noted, will both the docs and makefile targets as we have with dashboard.

**Comment by mfocko** [comment]
@majamassarini

1. I've tried setting up the local search, but hit some issues. On the other hand, they have an official support for 3rd party service that provides the search for free for open-source projects, but the site needs to be deployed for filling an application.
2. I'll try to play with it a bit more, I have some guesses as to why it is like that :/
3. I thought about it, though we have “abused” a lot the link `https://packit.dev/#contact` that goes straight to that paragraph on the front-page, so I'll probably add one to the footer, but keep the one on the front-page to make sure the link works as expected :smile: 

Thanks for the feedback :heart:

File: yarn.lock
Comment (majamassarini) [reply yarn.lock:1]: Oh I see 😅

**Comment by majamassarini** [comment]
> 1. I've tried setting up the local search, but hit some issues. On the other hand, they have an official support for 3rd party service that provides the search for free for open-source projects, but the site needs to be deployed for filling an application.

Ok!

> 3. I thought about it, though we have “abused” a lot the link `https://packit.dev/#contact` that goes straight to that paragraph on the front-page, so I'll probably add one to the footer, but keep the one on the front-page to make sure the link works as expected smile

A really good point!

File: README.md (hunk 2 above)
Comment (mfocko) [reply README.md:16]: Fixed by adding a makefile target `run-dev` that depends on `import` target that pull both research and deployment in

File: README.md (hunk 1 above)
Comment (mfocko) [reply README.md:10]: Fixed with the next point.

**Comment by mfocko** [comment]
As per the 2.:
image
Does it look better now? :D

**Review by lbarcziova — APPROVED** [review]
this is great, thanks! :medal_sports:  I like the additional features it provides us compared to the previous solution

maybe I would somehow merge Development, Deployment and Research, as these are all kinda development docs to me :thinking: as an outsider seeing e.g. the `Research` tab, I would be maybe confused...I am also wondering if these could be then somehow visually separated from the other "public-facing" categories

I'm ok with merging as it is and we can iterate more

**Comment by mfocko** [comment]
right, GH workflow is included…

**Pull request merged**